code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AutoCarouselNotificationBuilderTest.kt

```diff
@@ -0,0 +1,177 @@
+import com.adobe.marketing.mobile.notificationbuilder.internal.extensions.setRemoteViewClickAction
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.AutoCarouselPushTemplate
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.provideMockedAutoCarousalTemplate
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockkClass
+import io.mockk.mockkConstructor
+import io.mockk.mockkObject
+import io.mockk.verify
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [31])
+class AutoCarouselNotificationBuilderTest {
+
+    private lateinit var context: Context
+    private lateinit var expandedLayout: RemoteViews
+    private lateinit var trackerActivityClass: Class<out Activity>
+    private lateinit var broadcastReceiverClass: Class<out BroadcastReceiver>
+    private lateinit var autoCarouselPushTemplate: AutoCarouselPushTemplate
+
+    @Before
+    fun setUp() {
+        context = RuntimeEnvironment.getApplication()
+        expandedLayout = mockkClass(RemoteViews::class)
+        trackerActivityClass = mockkClass(Activity::class, relaxed = true).javaClass
+        broadcastReceiverClass = mockkClass(BroadcastReceiver::class, relaxed = true).javaClass
+        autoCarouselPushTemplate = provideMockedAutoCarousalTemplate(false)
+        mockkObject(BasicNotificationBuilder)
+        mockkObject(PendingIntentUtils)
+        mockkObject(PushTemplateImageUtils)
+        mockkConstructor(RemoteViews::class)
+    }
+
+    @Test
+    fun `construct returns NotificationCompat Builder for valid inputs`() {
+        val result = AutoCarouselNotificationBuilder.construct(
+            context,
+            autoCarouselPushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+        assertNotNull(result)
+    }
+
+    @Test
+    fun `construct returns BasicNotificationBuilder if less than 3 images were downloaded`() {
+        every { cacheImages(any()) } answers { 2 }
+        AutoCarouselNotificationBuilder.construct(
+            context,
+            autoCarouselPushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+        verify(exactly = 1) {
+            BasicNotificationBuilder.fallbackToBasicNotification(
+                context,
+                trackerActivityClass,
+                broadcastReceiverClass,
+                autoCarouselPushTemplate.data
+            )
+        }
+    }
+
+    @Test
+    fun `construct returns BasicNotificationBuilder if greater than equal to 3 images were downloaded`() {
```

Comment:
nit:`construct does not return BasicNotificationBuilder if greater than or equal to 3 images were downloaded`

Reply:
done @spoorthipujariadobe